packages/rich-text/src/component/index.js

```diff
@@ -137,6 +135,63 @@ function fixPlaceholderSelection( defaultView ) {
 	selection.collapseToStart();
 }
 
+function useReapply( {
+	TagName,
+	placeholder,
+	_value,
+	value,
+	record,
+	selectionStart,
+	selectionEnd,
+	isSelected,
+	remainingProps,
+	applyFromProps,
+} ) {
+	const shouldReapplyRef = useRef( false );
+
+	useEffect( () => {
+		shouldReapplyRef.current = true;
+	}, [ TagName, placeholder ] );
+
+	useEffect( () => {
+		if ( value !== _value.current ) {
+			shouldReapplyRef.current = true;
+		}
+	}, [ value, _value ] );
+
+	useEffect( () => {
+		const hasSelectionChanged =
+			selectionStart !== record.current.start ||
+			selectionEnd !== record.current.end;
+
+		if ( isSelected && hasSelectionChanged ) {
+			shouldReapplyRef.current = true;
+		} else if ( hasSelectionChanged ) {
+			record.current = {
+				...record.current,
+				start: selectionStart,
+				end: selectionEnd,
+			};
+		}
+	}, [ selectionStart, selectionEnd, isSelected, record ] );
+
+	const prefix = 'format_prepare_props_';
+	const predicate = ( v, key ) => key.startsWith( prefix );
+	const prepareProps = pickBy( remainingProps, predicate );
+
+	useShallowCompareEffect( () => {
+		shouldReapplyRef.current = true;
+	}, Object.values( prepareProps ) );
+
+	useEffect( () => {
+		if ( shouldReapplyRef.current ) {
+			applyFromProps();
+
+			shouldReapplyRef.current = false;
+		}
+	} );
```

Comment:
Missing dependencies?

Reply:
It's actually intended. No dependencies means that the effect will run after every render, so that we don't have to add `applyFromProps` to the dependency array. `applyFromProps` changes almost every render anyway, so it basically has the same effect.

Reply:
In that case it might be good to comment, since missing dependencies rare and others might be wondering if it's intentional or a mistake.

Reply:
What's wrong with calling `applyFromProps()` from the other effects?

Reply:
Yeah, a comment would be helpful!

If we want to call `applyFromProps` in other effects, it has to be in the dependency array  of that effect. However, since `applyFromProps` almost always changes, the effect will always fire too. One solution would be to split them up into different effects and register another effect to run `applyFromProps` (like this solution). Another solution would be to rewrite them to better match the mental modal of hooks, but that's more work and could potentially be a breaking change if we're not careful.